test/sqllogictest/index_planning.slt

```diff
@@ -91,26 +91,41 @@ create index foo_idx2 on foo(b, a);
 statement ok
 create index foo_idx3 on foo(b);
 
-#query T multiline
-#explain plan for select b, c from foo where a = 5 and b = 'this'
-#----
-#Project {
-#  outputs: [1, 2],
-#  Join {
-#    variables: [[(0, 1), (1, 0)], [(0, 0), (1, 1)]],
-#    ArrangeBy { keys: [[#1, #0]], Get { foo (u1) } },
-#    Constant [["this", 5]],
-#  }
-#}
-
-#TODO (wangandi) (materialize#616): uncomment these tests when FilterEqualLiteral is enabled
-#think about what the desired plan is when
-#statement ok
-#create index bar_idx on bar(a)
-#
-#query T multiline
-#explain plan for select foo.a, b, c, d, e from foo, bar where foo.a = bar.a and b = 'this'
-#----
+query T multiline
+explain plan for select b, c from foo where a = 5 and b = 'this'
+----
+Project {
+  outputs: [1, 2],
+  Join {
+    variables: [[(0, 0), (1, 1)], [(0, 1), (1, 0)]],
+    implementation: DeltaQuery,
+    ArrangeBy {
+      keys: [[#1, #0]],
+      Get { materialize.public.foo (u1) }
+    },
+    ArrangeBy { keys: [[#0, #1]], Constant [["this", 5]] }
+  }
+}
+
+statement ok
+create index bar_idx on bar(a)
+
+query T multiline
+explain plan for select foo.a, b, c, d, e from foo, bar where foo.a = bar.a and b = 'this'
+----
+Project {
+  outputs: [0 .. 2, 5, 6],
+  Join {
+    variables: [[(0, 0), (2, 0)], [(1, 0), (2, 1)]],
+    implementation: DifferentialLinear,
+    Get { materialize.public.bar (u3) },
+    ArrangeBy { keys: [[]], Constant [["this"]] },
```

Comment:
This is maybe an example of why we might want `ScalarExpr` in the join constraints. We want to introduce the constant into the expression, but doing so with a cross join will result in rough behavior.